app/controllers/service_provider_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ServiceProviderController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  def update
+    # TODO: shared secret token to limit authz to corresponding env dashboard
+
+    if FeatureManagement.use_dashboard_service_providers?
+      ServiceProviderUpdater.new.run
+      SecureHeadersWhitelister.new.run
+    end
+
+    # always return success in case feature is off
+    render json: { status: 'thanks' }
```

Comment:
I was going for non-committal, because of the security issue. Since this API endpoint is open to the world, I wanted a non-specific response so as not to reveal the presence/absence of the feature flag. I'll try something like what you suggest.